edx_management_commands/management_commands/management/commands/manage_user.py

```diff
@@ -0,0 +1,135 @@
+        """
+        old_value = getattr(user, attribute)
+        if new_value != old_value:
+            self.stderr.write(
+                _('Setting {attribute} for user "{username}" to "{new_value}"').format(
+                    attribute=attribute, username=user.username, new_value=new_value
+                )
+            )
+            setattr(user, attribute, new_value)
+
+    def _check_email_match(self, user, email):
+        """
+        DRY helper.
+
+        Requiring the user to specify both username and email will help catch
+        certain issues, for example if the expected username has already been
+        taken by someone else.
+        """
+        if user.email != email:
+            # The passed email address doesn't match this username's email address.
+            # Assume a problem and fail.
+            raise CommandError(
+                _(
+                    'Skipping user "{}" because the specified and existing email '
+                    'addresses do not match.'
+                ).format(user.username)
+            )
+
+    def _handle_remove(self, username, email):
+        try:
+            user = get_user_model().objects.get(username=username)
+        except get_user_model().DoesNotExist:
+            self.stderr.write(_('Did not find a user with username "{}" - skipping.').format(username))
+            return
+        self._check_email_match(user, email)
+        self.stderr.write(_('Removing user: "{}"').format(user))
+        user.delete()
+
+    @transaction.atomic
+    def handle(self, username, email, is_remove, is_staff, is_superuser, groups, *args, **options):
+
+        if is_remove:
+            return self._handle_remove(username, email)
+
+        old_groups, new_groups = [], []
+        user, created = get_user_model().objects.get_or_create(username=username)
+
+        if created:
+            user.set_unusable_password()
+            self.stderr.write(_('Created new user: "{}"').format(user))
+            self._maybe_update(user, 'email', email)
+        else:
+            # NOTE, we will not update the email address of an existing user.
+            self.stderr.write(_('Found existing user: "{}"').format(user))
+            self._check_email_match(user, email)
+            old_groups = list(user.groups.all())
+
+        self._maybe_update(user, 'is_staff', is_staff)
+        self._maybe_update(user, 'is_superuser', is_superuser)
+
+        # resolve the specified groups
+        for group_name in groups or []:
+
+            try:
+                group = Group.objects.get(name=group_name)  # pylint: disable=no-member
+                new_groups.append(group)
+            except Group.DoesNotExist:
+                # warn, but move on.
+                self.stderr.write(_('Could not find a group named "{}" - skipping.').format(group_name))
+
+        add_groups = [p for p in new_groups if p not in old_groups]
+        remove_groups = [p for p in old_groups if p not in new_groups]
```

Comment:
If you need to log the changes, you could just log `set(new_groups) - set(old_groups)` and `set(old_groups) - set(new_groups)`.